langmeta/langmeta/shared/src/main/scala/org/langmeta/semanticdb/internal/package.scala

```diff
@@ -10,132 +10,192 @@ import org.langmeta.inputs.{Input => dInput}
         } else {
-          dInput.VirtualFile(sfilename.toString, scontents)
+          dInput.VirtualFile(sfilename.toString, stext)
+        }
+      }
+      object sRange {
+        def unapply(srange: s.Range): Option[dPosition] = {
+          val dstartOffset = dinput.lineToOffset(srange.startLine) + srange.startCharacter
+          val dendOffset = dinput.lineToOffset(srange.endLine) + srange.endCharacter
+          Some(dPosition.Range(dinput, dstartOffset, dendOffset))
         }
       }
-      object sPosition {
-        def unapply(spos: s.Position): Option[dPosition] = {
-          Some(dPosition.Range(dinput, spos.start, spos.end))
+      object sRole {
+        def unapply(srole: s.SymbolOccurrence.Role): Option[Boolean] = srole match {
+          case s.SymbolOccurrence.Role.REFERENCE => Some(false)
+          case s.SymbolOccurrence.Role.DEFINITION => Some(true)
+          case _ => None
         }
       }
       object sSeverity {
-        def unapply(sseverity: s.Message.Severity): Option[d.Severity] = {
+        def unapply(sseverity: s.Diagnostic.Severity): Option[d.Severity] = {
           sseverity match {
-            case s.Message.Severity.INFO => Some(d.Severity.Info)
-            case s.Message.Severity.WARNING => Some(d.Severity.Warning)
-            case s.Message.Severity.ERROR => Some(d.Severity.Error)
+            case s.Diagnostic.Severity.ERROR => Some(d.Severity.Error)
+            case s.Diagnostic.Severity.WARNING => Some(d.Severity.Warning)
+            case s.Diagnostic.Severity.INFORMATION => Some(d.Severity.Info)
+            case s.Diagnostic.Severity.HINT => Some(d.Severity.Hint)
             case _ => None
           }
         }
       }
-      object sResolvedSymbol {
-        def unapply(sresolvedsymbol: s.ResolvedSymbol): Option[d.ResolvedSymbol] = sresolvedsymbol match {
-          case s.ResolvedSymbol(d.Symbol(dsym), Some(s.Denotation(dflags, dname: String, dsignature: String, snames, smembers, soverrides))) =>
-            val ddefninput = dInput.Denotation(dsignature, dsym)
-            val dnames = snames.toIterator.map {
-              case s.ResolvedName(Some(s.Position(sstart, send)), d.Symbol(dsym), disDefinition) =>
-                val ddefnpos = dPosition.Range(ddefninput, sstart, send)
-                d.ResolvedName(ddefnpos, dsym, disDefinition)
-              case other =>
-                sys.error(s"bad protobuf: unsupported name $other")
-            }.toList
+      object sSymbolInformation {
+        def unapply(ssymbolInformation: s.SymbolInformation): Option[d.ResolvedSymbol] = ssymbolInformation match {
+          case s.SymbolInformation(d.Symbol(dsym), _, skind, sproperties, sname, _, ssignature, smembers, soverrides) =>
+            val dflags = {
+              var dflags = 0L
+              def dflip(dbit: Long) = dflags ^= dbit
+              skind match {
+                case s.SymbolInformation.Kind.UNKNOWN1 => ()
```

Comment:
I'm surprised you haven't bound `val k = s.SymbolInformation.Kind` 😜

Reply:
Done.